### PR TITLE
feat: add optional ssz type

### PR DESCRIPTION
**Pull request by g11tech**
implement PR:
 - [eip-6475](https://eips.ethereum.org/EIPS/eip-6475)

requirement to build verkle types as the kaustenine network (and its relaunch will use optional types)

### CODE REVIEW

File: packages/ssz/src/type/optional.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    }
+  }
+
+  getPropertyType(): Type<unknown> {
+    return this.elementType;
+  }
+
+  getIndexProperty(index: number): string | number | null {
+    if (isCompositeType(this.elementType)) {
+      return this.elementType.getIndexProperty(index);
```
Comment (g11tech) [inline packages/ssz/src/type/optional.ts:169]: may be an de-concatenation op is required (to remove `VALUE_GINDEX` before passing in the index to the underlying element type

Comment (wemeetagain) [reply packages/ssz/src/type/optional.ts:170]: I think this method should only return `"value"` or `"selector"` for 0 or 1?

Comment (g11tech) [reply packages/ssz/src/type/optional.ts:170]: transparent because `value` and `selector` properties don't exist as per my understanding

**Comment by wemeetagain** [comment]
@g11tech I did a review of this PR and added some commits on top, let me know what you think.

**Review by wemeetagain — APPROVED** [review]

File: packages/ssz/src/type/optional.ts [hunk 2]
```diff
@@ -55,24 +59,34 @@ export class OptionalType<ElementType extends Type<unknown>> extends CompositeTy
     return null as ValueOfType<ElementType>;
   }
 
-  getView(tree: Tree): ValueOfType<ElementType> {
-    return this.tree_toValue(tree.rootNode);
+  // TODO fix
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  getView(_tree: Tree): ValueOfType<ElementType> {
+    throw new Error("not implemented");
```
Comment (g11tech) [inline packages/ssz/src/type/optional.ts]: @wemeetagain should we not depending on its composite type or not provide the toValue or underlying view?

**Comment by g11tech** [comment]
have merged for now @wemeetagain , we can try adding other view/tree methods if there is need :+1: 

Also as per the plan, the executionWitness might get serialized later on for embedding in executionPayload so we might not need this at all  (but i guess ethereumjs and/or other EL side codebases would be able to use this, so good for ecosystem in general)

**Pull request merged**